monkey/monkey_island/cc/resources/propagation_credentials.py

```diff
@@ -29,7 +32,8 @@ def get(self, collection=None):
 
         return propagation_credentials, HTTPStatus.OK
 
-    # Used by Agent. Can't secure.
+    @auth_token_required
+    @roles_accepted(AccountRole.AGENT.name, AccountRole.ISLAND_INTERFACE.name)
```

Comment:
Do agents actually need to put propagation credentials?

Reply:
I suppose they don't, since they send them via events.

Reply:
Good catch